core/trino-main/src/main/java/io/trino/sql/gen/columnar/InColumnarFilterGenerator.java

```diff
@@ -112,24 +131,62 @@ public InColumnarFilterGenerator(SpecialForm specialForm, FunctionManager functi
                 .collect(toImmutableList());
 
         checkArgument(specialForm.functionDependencies().size() == 3);
-        ResolvedFunction resolvedEqualsFunction = specialForm.getOperatorDependency(EQUAL);
-        ResolvedFunction resolvedHashCodeFunction = specialForm.getOperatorDependency(HASH_CODE);
         ResolvedFunction resolvedIsIndeterminate = specialForm.getOperatorDependency(INDETERMINATE);
-        equalsMethodHandle = functionManager.getScalarFunctionImplementation(resolvedEqualsFunction, simpleConvention(NULLABLE_RETURN, NEVER_NULL, NEVER_NULL)).getMethodHandle();
-        hashCodeMethodHandle = functionManager.getScalarFunctionImplementation(resolvedHashCodeFunction, simpleConvention(FAIL_ON_NULL, NEVER_NULL)).getMethodHandle();
         MethodHandle indeterminateMethodHandle = functionManager.getScalarFunctionImplementation(resolvedIsIndeterminate, simpleConvention(FAIL_ON_NULL, NEVER_NULL)).getMethodHandle();
-
         ImmutableSet.Builder<Object> constantValuesBuilder = ImmutableSet.builder();
         for (ConstantExpression testValue : testExpressions) {
             if (isDeterminateConstant(testValue, indeterminateMethodHandle)) {
                 constantValuesBuilder.add(testValue.value());
             }
         }
-        constantValues = constantValuesBuilder.build();
-        useSwitchCase = useSwitchCaseGeneration(valueExpression.type(), expressions);
+        Set<Object> constantValues = constantValuesBuilder.build();
+
+        if (constantValues.size() > 1 && isIntegerArithmeticValidType(valueExpression.type())) {
```

Comment:
We should move/copy this to the new IR expression optimizer framework.  I know this stuff is still using the row expressions but in the log run, when row expression is gone, I'd like to have this in the main IR optimizer

Reply:
Added a new IrOptimizerRule to handle this and removed the changes from here